### PR TITLE
[clang-format] Fix misannotations of `<` in ternary expressions

### LINKED ISSUE #100300: [clang-format] nested ternary misformatted in version 19
While trying out clang-format on the new 19.x branch, I discovered an issue with the formatting with a nested ternary.
I came up with the following minimal reproducer.

In clang-format 18.1.7 with the default LLVM style:
`bool foo(int A, int B) { return A < B ? true : A > B ? false : false; }`

In clang-format 19.1.0 (183e8ec)
`bool foo(int A, int B) { return A<B ? true : A> B ? false : false; }`

### PR DESCRIPTION
**Pull request by owenca**
Fixes #100300.

### CODE REVIEW

File: clang/lib/Format/TokenAnnotator.cpp [hunk 1]
```diff
@@ -176,31 +176,38 @@ class AnnotatingParser {
...
+        if (Next && Next->is(tok::greater) &&
             Left->ParentBracket != tok::less &&
             CurrentToken->getStartOfNonWhitespace() ==
-                CurrentToken->Next->getStartOfNonWhitespace().getLocWithOffset(
-                    -1)) {
+                Next->getStartOfNonWhitespace().getLocWithOffset(-1)) {
+          return false;
+        }
+        if (InExpr && SeenTernaryOperator &&
+            (!Next || Next->isNot(tok::l_paren))) {
```
Comment (rymiel) [inline clang/lib/Format/TokenAnnotator.cpp:210]: Maybe I'm looking too hard for edge-cases, but maybe we could also add l_brace here? This is valid code:

```c++
template <int I>
struct Foo {
    int i = I;
};

auto a = Foo<false ? 1 : 2>{};
```

**Review by rymiel — APPROVED** [review]
Wow!

**Comment by alanzhao1** [comment]
FYI this change messes up formatting of angle brackets for member function template parameters.

See #123144

**Comment by owenca** [comment]
See https://github.com/llvm/llvm-project/issues/123144#issuecomment-2597430287.

**Pull request merged**